lightning/src/ln/peer_handler.rs

```diff
@@ -354,7 +362,11 @@ pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: De
 	peer_counter_low: AtomicUsize,
 	peer_counter_high: AtomicUsize,
 
+	// Optional, if the user wishes to be notified about changes to the peer set
+	peer_connectivity_updater: Option<Box<dyn PeerConnectivityUpdater>>,
```

Comment:
For consistency with our codebase, should we use a template parameter? May also help resolve the compilation error:

```term
error: future cannot be sent between threads safely
   --> lightning-net-tokio/src/lib.rs:274:8
    |
274 |         Some(tokio::spawn(async move {
    |              ^^^^^^^^^^^^ future created by async block is not `Send`
    | 
   ::: /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.8.1/src/task/spawn.rs:127:21
    |
127 |         T: Future + Send + 'static,
    |                     ---- required by this bound in `tokio::spawn`
    |
    = help: the trait `Send` is not implemented for `(dyn PeerConnectivityUpdater + 'static)`
note: captured value is not `Send`
   --> lightning-net-tokio/src/lib.rs:292:8
    |
292 | ...                   peer_manager.socket_disconnected(&SocketDescriptor::new(Arc::clone(&us)));
    |                       ^^^^^^^^^^^^ has type `Arc<PeerManager<SocketDescriptor, Arc<CMH>, Arc<RMH>, Arc<L>>>` which is not `Send`
```